x/dualstaking/keeper/keeper.go

```diff
@@ -92,7 +93,7 @@ func (k Keeper) ChangeDelegationTimestampForTesting(ctx sdk.Context, provider, d
 	if !found {
 		return fmt.Errorf("cannot change delegation timestamp: delegation not found. provider: %s, delegator: %s", provider, delegator)
 	}
-	d.Timestamp = timestamp
```

Comment:
i don't get this comment, you need to update the delegation timestamp this is what this function does
you want the testing functions to use the same functions as the original code and not hack around them by implementing the same internal functionality
so even if you can do the same change more effectively you don't do that because that's not the purpose of tests

Reply:
I meant that after this change, the function looks like this:
```
func (k Keeper) ChangeDelegationTimestampForTesting(ctx sdk.Context, provider, delegator string, timestamp int64) error {
	d, found := k.GetDelegation(ctx, provider, delegator)
	if !found {
		return fmt.Errorf("cannot change delegation timestamp: delegation not found. provider: %s, delegator: %s", provider, delegator)
	}
	ctx = ctx.WithBlockTime(time.Unix(timestamp, 0))
	return k.SetDelegation(ctx, d)
}
```

So you get the delegation, change the ctx block time and then set the delegation. So getting and setting the delegation is not doing anything since you're not changing the delegation. Am I missing something?

Reply:
You're missing the change to the delegation timestamp